src/dashboard/Settings/components/SettingsLayout/index.js

```diff
@@ -0,0 +1,143 @@
+ *
+ * @function
+ * @name SettingsLayout
+ * @param {Object} props
+ * @param {string} props.incidentType
+ *
+ * @version 0.1.0
+ * @since 0.1.0
+ */
+
+class SettingsLayout extends React.Component {
+  onSearch = searchValue => {
+    const { searchIncidentTypeValue } = this.props;
+    searchIncidentTypeValue(searchValue);
+  };
+
+  render() {
+    const { incidentType } = this.props;
+
+    return (
+      <Fragment>
+        <Layout
+          style={{
+            background: '#fff',
+            borderTop: '1px solid #e6e6e6',
+          }}
+        >
+          <Row>
+            <Col span={4} className={cx('section')}>
+              <ColHeader className={cx('SettingLayoutContentHeader')} />
+              <Content className={cx('SettingLayoutContent')}>
+                <SidebarSettings title="System" />
+              </Content>
+            </Col>
+            <Col span={6} className={cx('section')}>
+              <ColHeader className={cx('SettingLayoutContentHeader')}>
+                <Row type="flex" justify="space-around">
+                  <Col span={1}>
+                    <Checkbox />
+                  </Col>
+                  <Col span={19}>
+                    <Search
+                      placeholder="Search here"
+                      style={{ width: '100%' }}
+                      onSearch={value => this.onSearch(value)}
+                      enterButton={<Button icon="search" />}
+                    />
+                  </Col>
+                </Row>
+              </ColHeader>
+              <Content className={cx('SettingLayoutContent')}>
+                <IncidentType />
+              </Content>
+            </Col>
+            <Col span={14} className={cx('section')}>
+              <ColHeader className={cx('SettingLayoutContentHeader')}>
+                <Row>
+                  <Col span={1}>
+                    <AddNewIncidentType />
+                  </Col>
+                  <Col span={1}>
+                    <EditIncidentTypeForm incidentType={incidentType[0]} />
+                  </Col>
+                  <Col span={21}>
+                    <h3>Basic Information</h3>
+                  </Col>
+                </Row>
+              </ColHeader>
+              <Content className={cx('SettingLayoutContent')}>
+                <IncidentDetails incidentType={incidentType} />
+              </Content>
+            </Col>
+          </Row>
+        </Layout>
+      </Fragment>
+    );
+  }
+}
+
+const mapStateToProps = state => ({
+  incidentsType: state.incidentsType.data,
+  incidentType: state.incidentsType.incidentType
+    ? [state.incidentsType.incidentType]
+    : [],
+});
+
+const mapDispatchToProps = dispatch => ({
+  searchIncidentTypeValue: bindActionCreators(searchIncidentType, dispatch),
+});
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(SettingsLayout);
+
+/* props validations for SettingsLayout */
+const incidentTypePropTypes = PropTypes.shape({
```

Comment:
Props validations for class(stateful component) should be inside class

Reply:
cool